base/sparse/sparsematrix.jl

```diff
@@ -1564,10 +1602,249 @@ function _map_notzeropres!{Tf}(f::Tf, fillvalue, C::SparseMatrixCSC, A::SparseMa
     end
     return C
 end
+# _broadcast_zeropres!/_broadcast_notzeropres! specialized for a pair of (input) sparse matrices
+function _broadcast_zeropres!{Tf}(f::Tf, C::SparseMatrixCSC, A::SparseMatrixCSC, B::SparseMatrixCSC)
+    isempty(C) && (fill!(C.colptr, 1); return C)
+    spaceC = min(length(C.rowval), length(C.nzval))
+    rowsentinelA = convert(eltype(A.rowval), C.m + 1)
```

Comment:
> I suppose rowsentinelA::eltype(A.rowval) = C.m + 1 might provide the best of both worlds?

A further thought: A few assertions like `Ak::eltype(A.rowval)` might allow us to simplify some expressions, e.g. `Ak += one(Ak)` -> `Ak += 1`. So perhaps adopting that style uniformly in these methods would be advantageous?